Demo/DirectoryWatcher/DirectoryWatcherTest.cpp

```diff
@@ -0,0 +1,8 @@
+#include <iostream>
```

Comment:
1. Make this program to accept arguments. It should get a file path and start watching over the file. 
2. Parse those arguments with argument parser.
3. Ask Chat-GPT to generate a bash script which creates a testing infrastructure (directory with subdirectories with subdirectories with random regular files, or just a file), starts watching over the infrastructure using your program, makes changes over the infrastructure, collects the logs and compares with the expected output. Don't forget to remove this mess.